packages/thirdweb/src/wallets/eip5792/get-calls-status.ts

```diff
@@ -0,0 +1,90 @@
+import type { ThirdwebClient } from "../../client/client.js";
+import { isCoinbaseSDKWallet } from "../coinbase/coinbaseSDKWallet.js";
+import { isInAppWallet } from "../in-app/core/wallet/index.js";
+import { getInjectedProvider } from "../injected/index.js";
+import type { Wallet } from "../interfaces/wallet.js";
+import { isSmartWallet } from "../smart/index.js";
+import { isWalletConnect } from "../wallet-connect/index.js";
+import type { GetCallsStatusResponse, WalletSendCallsId } from "./types.js";
+
+export type GetCallsStatusOptions = {
+  wallet: Wallet;
+  client: ThirdwebClient;
+  bundleId: WalletSendCallsId;
+};
+
+/**
+ * Get the status of an [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792) bundle.
+ *
+ * Note: This function is dependent on the wallet's support for EIP-5792 and could fail.
+ *
+ * @param {GetCallsStatusOptions} options
+ * @param {Wallet} options.wallet - The wallet that send the original calls.
+ * @param {ThirdwebClient} options.client - A {@link ThirdwebClient} instance.
+ * @param {WalletSendCallsId} options.bundleId - The ID of the bundle to get the status of.
+ * @throws an error if the wallet does not support EIP-5792.
+ * @returns {Promise<GetCallsStatusResponse>} - A promise that resolves to the bundle's status and receipts (if available). {@link GetCallsStatusResponse}
+ *
+ * @example
+ * ```ts
+ *  import { createThirdwebClient } from "thirdweb";
+ *  import { sendCalls, getCallsStatus } from "thirdweb/wallets";
+ *
+ *  const client = createThirdwebClient({ clientId: ... });
+ *
+ *  const bundleId = await sendCalls({ wallet, client, calls });
+ *
+ *  let result;
+ *  while (result.status !== "CONFIRMED") {
+ *    result = await getCallsStatus({ wallet, client, bundleId });
+ *  }
+ * ```
+ * @wallets
+ */
+export async function getCallsStatus({
+  wallet,
+  client,
+  bundleId,
+}: GetCallsStatusOptions): Promise<GetCallsStatusResponse> {
+  const account = wallet.getAccount();
+  if (!account) {
+    throw new Error(
+      `Failed to get call status, no account found for wallet ${wallet.id}`,
+    );
+  }
+
+  // These conveniently operate the same
+  if (isSmartWallet(wallet) || isInAppWallet(wallet)) {
```

Comment:
feels like it would be much cleaner to have a `getCallStatus` (or any of the new methods) as optional methods in the wallet interface.

then this function just becomes:

```
// if wallet.getCallStatus() => try to call it
// else throw with not supported
```

you're already effectively doing a diff implementation of each new function for each wallet type, much more suited to be in wallet interface.